docs/source/1.0/guides/generating-cloudformation-resources.rst

```diff
@@ -0,0 +1,484 @@
+    result as needed.
+
+    .. code-block:: json
+
+        {
+            "version": "1.0",
+            "plugins": {
+                "cloudformation": {
+                    "service": "smithy.example#Queues",
+                    "organizationName": "Smithy",
+                    "jsonAdd": {
+                        "/info/title": "Replaced title value",
+                        "/info/nested/foo": {
+                            "hi": "Adding this object created intermediate objects too!"
+                        },
+                        "/info/nested/foo/baz": true
+                    }
+                }
+            }
+        }
+
+.. _generate-cloudformation-setting-disableDeprecatedPropertyGeneration:
+
+disableDeprecatedPropertyGeneration (``boolean``)
+    Sets whether to disable generating ``deprecatedProperties`` for Resource
+    Schemas. By default, deprecated members are automatically added to the
+    ``deprecatedProperties`` schema property.
+
+    .. code-block:: json
+
+        {
+            "version": "1.0",
+            "plugins": {
+                "cloudformation": {
+                    "service": "smithy.example#Queues",
+                    "organizationName": "Smithy",
+                    "disableDeprecatedPropertyGeneration": true
+                }
+            }
+        }
+
+.. _generate-cloudformation-setting-disableCapitalizedProperties:
+
+disableCapitalizedProperties (``boolean``)
+    Sets whether to disable automatically capitalizing names of properties of
+    Resource Schemas. By default, property names of resource schemas are
+    capitalized if no :ref:`cfnName <aws.cloudformation#cfnName-trait>` trait
+    is applied.
+
+    .. code-block:: json
+
+        {
+            "version": "1.0",
+            "plugins": {
+                "cloudformation": {
+                    "service": "smithy.example#Queues",
+                    "organizationName": "Smithy",
+                    "disableCapitalizedProperties": true
+                }
+            }
+        }
+
+----------------------------------
+JSON schema configuration settings
+----------------------------------
+
+.. _generate-cloudformation-jsonschema-setting-defaultTimestampFormat:
+
+defaultTimestampFormat (``string``)
+    Sets the assumed :ref:`timestampFormat-trait` value for timestamps with
+    no explicit timestampFormat trait. The provided value is expected to be
+    a string. Defaults to "date-time" if not set. Can be set to "date-time",
+    "epoch-seconds", or "http-date".
+
+    .. code-block:: json
+
+        {
+            "version": "1.0",
+            "plugins": {
+                "cloudformation": {
+                    "service": "smithy.example#Queues",
+                    "organizationName": "Smithy",
+                    "defaultTimestampFormat": "epoch-seconds"
+                }
+            }
+        }
+
+.. _generate-cloudformation-jsonschema-setting-unionStrategy:
+
+unionStrategy (``string``)
```

Comment:
Is this something that we can just pick for everyone? Or does it need to be customizable?

Reply:
I think this can be restricted down to `oneOf` safely, updating.